xumm/xapp/post_xapp_push_test.go

```diff
@@ -0,0 +1,115 @@
+package xapp
+
+import (
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	anyjson "github.com/xyield/xumm-go-client/pkg/json"
+	testutils "github.com/xyield/xumm-go-client/pkg/test-utils"
+	"github.com/xyield/xumm-go-client/xumm"
+	"github.com/xyield/xumm-go-client/xumm/models"
+)
+
+func TestPostXappPush(t *testing.T) {
+
+	var tests = []struct {
+		description    string
+		request        models.XappRequest
+		jsonRequest    string
+		jsonResponse   string
+		expectedOutput *models.XappResponse
+		expectedError  error
+		httpStatusCode int
+	}{
+		{
+			description: "successfully create push",
+			request: models.XappRequest{
+				UserToken: "token",
+				Subtitle:  "subtitle",
+				Body:      "body",
+				Data:      anyjson.AnyJson{},
+			},
+			jsonRequest: `{
+				"user_token": "token",
+				"subtitle": "subtitle",
+				"body": "body"
+			}`,
+			jsonResponse: `{
+				"pushed": true
+			  }`,
+			expectedOutput: &models.XappResponse{Pushed: true},
+			expectedError:  nil,
+			httpStatusCode: 200,
+		},
+		{
+			description: "invalid request body",
+			request: models.XappRequest{
+				UserToken: "",
+				Subtitle:  "",
+				Body:      "body",
+				Data:      anyjson.AnyJson{},
+			},
+			jsonRequest:    "",
+			jsonResponse:   "",
+			expectedOutput: nil,
+			expectedError:  &invalidPushRequestError{},
+			httpStatusCode: 200,
+		},
+		{
+			description: "error creating push",
+			request: models.XappRequest{
+				UserToken: "token",
+				Subtitle:  "subtitle",
+				Body:      "body",
+				Data:      anyjson.AnyJson{},
+			},
+			jsonRequest: `{
+				"user_token": "token",
+				"subtitle": "subtitle",
+				"body": "body"
+			}`,
+			jsonResponse: `{
+				"error": {
+				  "reference": "42d58b17-ee92-419d-b8ec-15797d10c4ed",
+				  "code": 400
+				}
+			  }`,
```

Comment:
Can we also add a test with a non empty data field to see that it serialises properly.

Reply:
added test